python/hsfs/core/feature_monitoring_config_engine.py

```diff
@@ -0,0 +1,521 @@
+#
+#   Copyright 2024 Hopsworks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+from typing import Any, Dict, List, Optional, Union
+from datetime import date, datetime
+import re
+import pandas as pd
+
+from hsfs.core import feature_monitoring_config_api
+from hsfs.core import feature_monitoring_config as fmc
+from hsfs.core.feature_monitoring_result_engine import FeatureMonitoringResultEngine
+from hsfs.core.feature_monitoring_result import FeatureMonitoringResult
+from hsfs.core import monitoring_window_config as mwc
+from hsfs.core import monitoring_window_config_engine
+
+from hsfs.core.job import Job
+from hsfs.core.job_api import JobApi
+from hsfs.client.exceptions import FeatureStoreException
+
+from hsfs import feature_group, feature_view
+
+
+class FeatureMonitoringConfigEngine:
+    def __init__(
+        self,
+        feature_store_id: int,
+        feature_group_id: Optional[int] = None,
+        feature_view_name: Optional[str] = None,
+        feature_view_version: Optional[int] = None,
+        **kwargs,
+    ):
+        """Business logic for feature monitoring configuration.
+
+        This class encapsulates the business logic for feature monitoring configuration.
+        It is responsible for routing methods from the public python API to the
+        appropriate REST calls. It should also contain validation and error handling logic
+        for payloads or default object. Additionally, it contains logic necessary
+        to run the feature monitoring job, including taking a monitoring window configuration
+        and fetching the associated data.
+
+        Attributes:
+            feature_store_id: int. Id of the respective Feature Store.
+            feature_group_id: int. Id of the feature group, if monitoring a feature group.
+            feature_view_name: str. Name of the feature view, if monitoring a feature view.
+            feature_view_version: int. Version of the feature view, if monitoring a feature view.
+        """
+        self._feature_store_id = feature_store_id
+        self._feature_group_id = feature_group_id
+        self._feature_view_name = feature_view_name
+        self._feature_view_version = feature_view_version
+
+        # if feature_group_id is None:
```

Comment:
If not needed please remove, or add a comment why to keep it

Reply:
Removed.